package/kernel/mac80211/patches/ath10k/990-ath10k-small-buffers.patch

```diff
@@ -1,64 +1,175 @@
++		ath10k_smallbuffers = false;
++		break;
++	default:
++		break;
++	}
++
++	return 0;
++}
++
+ int ath10k_core_start(struct ath10k *ar, enum ath10k_firmware_mode mode,
+ 		      const struct ath10k_fw_components *fw)
+ {
+@@ -3226,6 +3258,12 @@ int ath10k_core_start(struct ath10k *ar,
+ 	if (status)
+ 		goto err_hif_stop;
+ 
++	status = ath10k_core_check_smallbuffers();
++	if (status) {
++		ath10k_err(ar, "checking if driver should reduce buffers failed: %d\n", status);
++		goto err_hif_stop;
++	}
++
+ 	return 0;
+ 
+ err_hif_stop:
+--- a/drivers/net/wireless/ath/ath10k/core.h
++++ b/drivers/net/wireless/ath/ath10k/core.h
+@@ -889,6 +889,12 @@ enum ath10k_crypt_mode {
+ 	ATH10K_CRYPT_MODE_SW,
  };
  
- #define HTT_RX_RING_SIZE_MIN 128
-+#ifndef CONFIG_ATH10K_SMALLBUFFERS
- #define HTT_RX_RING_SIZE_MAX 2048
-+#else
-+#define HTT_RX_RING_SIZE_MAX 512
-+#endif
- #define HTT_RX_RING_SIZE HTT_RX_RING_SIZE_MAX
- #define HTT_RX_RING_FILL_LEVEL (((HTT_RX_RING_SIZE) / 2) - 1)
- #define HTT_RX_RING_FILL_LEVEL_DUAL_MAC (HTT_RX_RING_SIZE - 1)
---- a/drivers/net/wireless/ath/ath10k/pci.c
-+++ b/drivers/net/wireless/ath/ath10k/pci.c
-@@ -131,7 +131,11 @@ static const struct ce_attr pci_host_ce_
- 		.flags = CE_ATTR_FLAGS,
- 		.src_nentries = 0,
- 		.src_sz_max = 2048,
-+#ifndef CONFIG_ATH10K_SMALLBUFFERS
- 		.dest_nentries = 512,
-+#else
-+		.dest_nentries = 128,
-+#endif
- 		.recv_cb = ath10k_pci_htt_htc_rx_cb,
- 	},
++enum ath10k_smallbuffers_mode {
++	ATH10K_SMALLBUFFERS_AUTO,
++	ATH10K_SMALLBUFFERS_ON,
++	ATH10K_SMALLBUFFERS_OFF,
++};
++
+ static inline const char *ath10k_cal_mode_str(enum ath10k_cal_mode mode)
+ {
+ 	switch (mode) {
+@@ -1328,6 +1334,7 @@ static inline bool ath10k_peer_stats_ena
  
-@@ -140,7 +144,11 @@ static const struct ce_attr pci_host_ce_
- 		.flags = CE_ATTR_FLAGS,
- 		.src_nentries = 0,
- 		.src_sz_max = 2048,
-+#ifndef CONFIG_ATH10K_SMALLBUFFERS
- 		.dest_nentries = 128,
-+#else
-+		.dest_nentries = 64,
-+#endif
- 		.recv_cb = ath10k_pci_htc_rx_cb,
- 	},
+ extern unsigned int ath10k_frame_mode;
+ extern unsigned long ath10k_coredump_mask;
++extern bool ath10k_smallbuffers;
  
-@@ -167,7 +175,11 @@ static const struct ce_attr pci_host_ce_
- 		.flags = CE_ATTR_FLAGS,
- 		.src_nentries = 0,
- 		.src_sz_max = 512,
-+#ifndef CONFIG_ATH10K_SMALLBUFFERS
- 		.dest_nentries = 512,
-+#else
-+		.dest_nentries = 128,
-+#endif
- 		.recv_cb = ath10k_pci_htt_rx_cb,
- 	},
+ void ath10k_core_napi_sync_disable(struct ath10k *ar);
+ void ath10k_core_napi_enable(struct ath10k *ar);
+--- a/drivers/net/wireless/ath/ath10k/htt_rx.c
++++ b/drivers/net/wireless/ath/ath10k/htt_rx.c
+@@ -792,7 +792,10 @@ int ath10k_htt_rx_alloc(struct ath10k_ht
+ 	/* XXX: The fill level could be changed during runtime in response to
+ 	 * the host processing latency. Is this really worth it?
+ 	 */
+-	htt->rx_ring.size = HTT_RX_RING_SIZE;
++	if (ath10k_smallbuffers)
++		htt->rx_ring.size = 512;
++	else
++		htt->rx_ring.size = HTT_RX_RING_SIZE;
```

Comment:
think for this simple if we can use something like

`ath10k_buffers_mode == ...REDUCED : 512 ? ...RING_SIZE;`